csharp/src/SeedLang/Engine.cs

```diff
@@ -97,19 +101,20 @@ public class Engine {
 
     // Dumps the AST tree of the source code.
     public string DumpAst() {
-      Debug.Assert(!(_astTree is null));
+      Debug.Assert(!(_astTree is null), "Parse and compile source code before dumping AST tree.");
```

Comment:
Given `DumpAst` is a public interface exposed to SeedLang users, it's better to return the error message (via Diagnostics) directly, instead of asserting a prediction.

Reply:
Fixed.